lbr/lbr.py

```diff
@@ -220,30 +220,48 @@ def mrn_cond(l):
       if C.any_in(vecs, ''.join(v2ii2v_srcs)): inc_stat('V2I transition-Penalty')
       elif C.any_in(vecs, v2ii2v_dst): inc_stat('I2V transition-Penalty')
     elif v2ii2v_dst: inc_stat('V2I transition-Penalty')
-  if size > 1:
-    xline = LC.prev_line(lines)
-    xinfo = LC.line2info(xline)
-    if 'dsb-heatmap' in hsts and (xinfo.is_taken() or new_line):
-      inc(hsts['dsb-heatmap'], pmu.dsb_set_index(ip))
-    if 'indirect-x2g' in hsts and LC.is_type(x86.INDIRECT, xline):
-      ilen = xinfo.ilen() or 2
-      if abs(ip - (xip + ilen)) >= 2 ** 31:
-        inc(hsts['indirect-x2g'], xip)
-        if 'MISP' in xline: inc(hsts['indirect-x2g-misp'], xip)
-    if xip and xip in indirects:
-      inc(hsts['indirect_%s_targets' % LC.hex_ip(xip)], ip)
-      if 'MISP' in xline: inc(hsts['indirect-misp_%s_targets' % LC.hex_ip(xip)], ip)
-      #inc(hsts['indirect_%s_paths' % hex_ip(xip)], '%s.%s.%s' % (hex_ip(get_taken(lines, -2)['from']), hex_ip(xip), hex_ip(ip)))
-    if xinfo.is_cond_br() and xinfo.is_taken():
-      LC.glob['cond_%sward-taken' % ('for' if ip > xip else 'back')] += 1
-    # checks all lines but first
-    if info.is_cond_br():
-      if info.is_taken(): LC.glob['cond_taken-not-first'] += 1
-      else: LC.glob['cond_non-taken'] += 1
-      if x86_f.is_jcc_fusion(xline, line):
-        LC.glob['cond_fusible'] += 1
-        if size > 2 and xinfo.is_test_cmp() and LC.is_type(x86.LOAD, LC.prev_line(lines, step=2)):
-          inc_pair('LD-CMP', suffix='fusible')
+  if xinfo.is_cond_br() and xinfo.is_taken():
+    LC.glob['cond_%sward-taken' % ('for' if ip > xip else 'back')] += 1
+  #Shortcircuit detection
+  #A setcc instructions is found, now walk in reverse and find at least two conditional jumps within a max_inst window.
```

Comment:
Beyond this 1 lines, Would help the reader to have an example of a sequence of instructions the logic would catch